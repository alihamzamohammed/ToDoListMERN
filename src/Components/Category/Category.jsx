import React from "react";
import { Row, Col } from "react-bootstrap";
import { Link } from "react-router-dom";
import Todo from "../Todo/Todo";

const Category = (props) => {
  return (
    <Link to={"/edit/category/" + props.id} className="link-text">
      <div className="category">
        <Row>
          <Col xmd="1" className="category-id">
            <h4>{props.id}</h4>
          </Col>
        </Row>
        <Row>
          <Col xs="12" className="category-name">
            <h3>{props.name}</h3>
          </Col>
        </Row>
        <Row>
          {props.todos.map((todo, idx) => (
            <Todo
              title={todo.title}
              content={todo.content}
              id={todo._id}
              key={idx}
            />
          ))}
        </Row>
      </div>
    </Link>
  );
};

export default Category;
