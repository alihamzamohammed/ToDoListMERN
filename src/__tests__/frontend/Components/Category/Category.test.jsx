import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";

import { BrowserRouter as Router } from "react-router-dom";
import Category from "../../../../Components/Category/Category";

describe("Todo Card Tests", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  it("Renders the navbar", () => {
    act(() => {
      render(
        <Router>
          <Category name="Test Name" id="1234" todos={[]}></Category>
        </Router>,
        container
      );
    });
    expect(container).toMatchSnapshot();
    expect(container.querySelector(".category-name").textContent).toBe(
      "Test Name"
    );
    expect(container.querySelector(".category-id").textContent).toBe("1234");
  });
});
