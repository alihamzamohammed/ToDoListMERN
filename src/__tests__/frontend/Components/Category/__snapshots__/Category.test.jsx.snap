// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo Card Tests Renders the navbar 1`] = `
<div>
  <a
    class="link-text"
    href="/edit/category/1234"
  >
    <div
      class="category"
      id="category-1234"
    >
      <div
        class="row"
      >
        <div
          class="category-id col"
          xmd="1"
        >
          <h4>
            1234
          </h4>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="category-name col-12"
        >
          <h3>
            Test Name
          </h3>
        </div>
      </div>
      <div
        class="row"
      />
    </div>
  </a>
</div>
`;
